diagnostics/info.sh

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env bash
+_modules=('system' 'bat' 'bat_config' 'bat_wrapper' 'tool')
+_modules_consented=()
+
+
+# -----------------------------------------------------------------------------
+# Modules:
+# -----------------------------------------------------------------------------
+
+_bat_:description() {
+	_collects "Version information for 'bat'."
+}
+
+_bat_config_:description() {
+	_collects "The environment variables used by 'bat'."
+	_collects "The 'bat' configuration file."
+}
+
+_bat_wrapper_:description() {
+	_collects "Any wrapper script used by 'bat'."
+}
+
+_system_:description() {
+	_collects "Operating system name."
+	_collects "Operating system version."
+}
+
+_tool_:description() {
+	_collects "Version information for 'less'."
+}
+
+
+# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+_bat_:run() {
+	_out bat --version
+	_out env | grep '^BAT_\|^PAGER='
+}
+
+_bat_config_:run() {
+	if [[ -f "$(bat --config-file)" ]]; then 
+		_out cat "$(bat --config-file)"; 
+	fi
+}
+
+_bat_wrapper_:run() {
+	if file "$(which bat)" | grep "text executable" &>/dev/null; then
+		_out cat "$(which bat)"
+	fi
+}
+
+_system_:run() {
+	_out uname -srm
+
+	if command -v "sw_vers" &>/dev/null; then _out sw_vers; fi
+	if command -v "lsb_release" &>/dev/null; then _out lsb_release -a; fi
+}
+
+_tool_:run() {
+	_out less --version | head -n1
+}
+
+
+# -----------------------------------------------------------------------------
+# Functions:
+# -----------------------------------------------------------------------------
+
+_out() {
+	printf "\n+" 1>&2
+	printf " %s" "$@" 1>&2
+	printf "\n" 1>&2
+
+	"$@" 2>&1
+}
+
+_tput() {
+	tput "$@" 1>&2 2>/dev/null
+}
+
+_collects() {
+	printf " - %s\n" "$1" 1>&2
+}
+
+_ask_module() {
+	_tput clear
+	_tput cup 0 0
+
+cat 1>&2 <<EOF
+--------------------------------------------------------------------------------
+This script will collect information to help assist in issue resolution.
+You will be provided with a description of the data collected, and will be
+offered the opportunity to give consent for each piece of information collected.
```

Comment:
I really appreciate this. However, it almost sounds a little bit too "dramatic". I wonder if someone could even get the impression that this script will directly upload the information somewhere (and do some magic things to "resolve issues").

I would propose something more straightforward along the lines of:
* this script runs some (harmless) commands to collect information about your system
* It will ask for consent before each command is executed
* In the end, it will output a short report that you can review and then copy into the issue description.

what do you think?

Reply:
Sounds good! I'll add it to the next commit